WordPress/Classes/Services/PeopleService.swift

```diff
@@ -119,33 +129,47 @@ struct PeopleService {
     /// - Returns: A new Person instance, with the new Role already assigned.
     ///
     func updateUser(_ user: User, role: String, failure: ((Error, User) -> Void)?) -> User {
```

Comment:
I'm a bit taken aback by this method having a return value as well as an optional failure callback. It makes the implementation code confusing, IMHO.

As far as I can see, it's used [only in one place](https://github.com/wordpress-mobile/WordPress-iOS/blob/230140d8449c981859235a6a7c1f2cb926e4d7f6/WordPress/Classes/ViewRelated/People/PersonViewController.swift#L376-L402). From reading the code, I'm guessing the method signature was dictated by the will to optimistically update the UI. Still, we could enable the caller to do that by using only callbacks. E.g.

```swift
updateUser(
  user,
  role: newRole,
  optimisticSuccess: { /* update UI but keep showing loading indicator */ },
  success: { /* remove loading indicator */ },
  failure: { /* revert UI and show error */ }
)
```

Also, I'm not sure optimistically updating the UI only to then show an out of context alert if the user has moved onto a different screen is doing them a good service. It would be best to subtly suggest that the change they requested is in progress, so that a possible failure wouldn't come across as abrupt.

What I'm suggesting is up for debate and out of scope for this PR... But it came to mind when reading the code and wanted to track it _somewhere_.

Reply:
I agree with what you raised here. But as you mentioned, I probably don't have the capacity to handle the issue right now. What do you think we raise this as an issue, and hopefully it gets picked up one day, but at least there is a dedicated place to discuss what's an appropriate UX in context of this "people management flow"?

Reply:
Sure. To be fair, I should have been the one opening the issue after that comment 🤦‍♂️ 

It might be a good trampoline to revisit how and where we do optimistic UI updates at the app level.